server/mediaserver.go

```diff
@@ -592,12 +591,16 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 
 	now := time.Now()
 	mid := parseManifestID(r.URL.Path)
-	s.connectionLock.Lock()
+	if mid == "" {
+		http.Error(w, `Bad URL`, http.StatusBadRequest)
+		return
+	}
+	s.connectionLock.RLock()
```

Comment:
The `cxn.lastUsed = now` assignment is a write, not a read - need to keep the `Lock` here. (Yes, it is not ideal to take a global lock here but it's OK, for now to avoid introducing yet another layer of connection-level locking.)

Reply:
It doesn't mutate `rtmpConnections` map - which we're protecting by this lock. So why not to use `RLock`?

Reply:
Multiple requests to the same MID could still change the same object

Reply:
It's just one `lastUsed` field. Don't see any problem in that.